src/hiero_sdk_python/account/account_id.py

```
@@ -165,6 +283,67 @@ def to_string_with_checksum(self, client: "Client") -> str:
+            if not account_id:
+                raise ValueError(f"Mirror node response missing 'account': {data}")
+        
+        except RuntimeError as e:
+            raise RuntimeError(
+                f"Failed to populate account number from mirror node for evm_address "
+                f"{self.evm_address.to_string()}"
+            ) from e    
+        
+        try:
+            self.num = int(account_id.split(".")[-1])
+            return self
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)